crates/storage/provider/src/traits/block.rs

```diff
@@ -50,6 +52,58 @@ pub trait BlockWriter: Send + Sync {
     fn insert_block(&self, block: SealedBlockWithSenders)
         -> ProviderResult<StoredBlockBodyIndices>;
 
+    /// Inserts additional block data that is not part of the header or transactions.
+    ///
+    /// This function handles the insertion of ommers, withdrawals, and requests data.
+    ///
+    /// # Arguments
+    /// - `block` - The `SealedBlockWithSenders` containing the additional data to be inserted
+    /// - `durations_recorder` - A mutable reference to a `DurationsRecorder` for performance
+    ///   tracking
+    ///
+    /// # Returns
+    /// - `ProviderResult<()>` - Ok if the insertion was successful, or an error if it failed
+    ///
+    /// # Tables modified
+    /// - `BlockOmmers` - If the block contains ommers
+    /// - `BlockWithdrawals` - If the block contains withdrawals
+    /// - `BlockRequests` - If the block contains requests
+    fn insert_block_additional_data(
+        &self,
+        block: SealedBlockWithSenders,
+        durations_recorder: &mut metrics::DurationsRecorder,
```

Comment:
I also don't think this needs to accept a `&mut metrics::DurationsRecorder`, we can just initialize it as `DurationsRecorder::default()` in each method's implementation, similar to how we do it now in `insert_block`

Reply:
actually nevermind, this interface is correct